pyproject.toml

```
@@ -269,7 +269,7 @@ force-exclude = '''
 [tool.codespell]
 skip = "*.dat,*.fits,*.hdr,*.xml,*egg*,*extern,.git,.tox,_build,fitshdr.htest_groups.py,venv,_*.c,wcs.h,lin.h,tab.h,spc.h,cython*"
```

```suggestion
skip = "*.dat,*.fits,*.hdr,*.xml,*egg*,*extern/*,.git,.tox,_build,fitshdr.htest_groups.py,venv,_*.c,wcs.h,lin.h,tab.h,spc.h,cython*"
```

I have had a chance to do some digging into `codespell` on the issue of pre-commit not respecting the `skip` arguments, which resulted in adding the `exclude` to the `pre-commit-config.yaml`.

It turns out that `codespell` uses [`fnmatch.fnmatch`](https://docs.python.org/3/library/fnmatch.html#fnmatch.fnmatch) to match things passed in to things listed in the `skip` list. The match for `cextern` (and other `extern`) would occur at the directory level when `codespell` is parsing over the directory structure meaning `*extern` would indeed match and then it would not recurse to subdirectories. However when passing a file inside `cextern` directly, like `pre-commit` does, this results in the filename (which includes the file path) not matching `*extern` because we have not indicated a wildcard for suffix matching. This change adds a wildcard for the suffix which should cause the desired effect of files inside `cextern` to be properly skipped.

---

Thank you for digging into this!  Let's give it a try.

We have a saying over at PlasmaPy that:

> Sphinx rabbit holes often have dragons in them.

Maybe this saying can be applied to pre-commit as well? 👀 🐉 🕳️ 🐰